hail/python/hailtop/batch/batch.py

```diff
@@ -138,16 +138,15 @@ def __init__(self,
         self._default_python_image = default_python_image
 
         self._project = project
-        self.__gcs: Optional[GCS] = None
+        self.__fs: Optional[RouterAsyncFS] = None
 
         self._cancel_after_n_failures = cancel_after_n_failures
 
     @property
-    def _gcs(self):
-        if self.__gcs is None:
-            self.__gcs = GCS(blocking_pool=concurrent.futures.ThreadPoolExecutor(),
-                             project=self._project)
-        return self.__gcs
+    def _fs(self):
+        if self.__fs is None:
+            self.__fs = RouterAsyncFS('file', [LocalAsyncFS(ThreadPoolExecutor()), GoogleStorageAsyncFS(project=self._project)])
+        return self.__fs
```

Comment:
This needs to be closed. I think it needs to live on the backend, not the batch. The backend has a close method.

Reply:
I think this was addressed in #10562.